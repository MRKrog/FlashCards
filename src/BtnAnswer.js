import React, { Component } from 'react';

import './styles/BtnAnswer.scss';

class BtnAnswer extends Component {
  constructor(props) {
    super(props);
  }

  handleChangeCard = () => {
    this.props.onChangeCard(this.props.answer);
  }

  render() {
    const cardAnswer = this.props.answer;

      return (
        <div className="BtnAnswer">
          <button onClick={this.handleChangeCard}>{cardAnswer}</button>
        </div>
      )
  }
}


export default BtnAnswer;
