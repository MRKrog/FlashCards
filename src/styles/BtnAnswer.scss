$font-title: 'Abel', sans-serif;
$font-copy: 'Lato', sans-serif;
$font-header: 'Barlow', sans-serif;

$white: #fff;
$lightWhite: #d6d6d6;
$black: #231f20;
$dark-blue: #11212a;

$purpleBtn: #723bd7;

.BtnAnswer {
  padding: .6em 0;
  position: relative;

    button {
      background: rgba(77, 27, 169, 0.65);
      color: rgba(255, 255, 255, .95);
      opacity: 1;
      width: 100%;
      font-family: $font-header;
      transition: all .3s ease;
      padding: .8em 0;
      cursor: pointer;
      border-radius: 2em;
      font-size: 1.1em;
      border: solid 1px #6a21ab;
      box-shadow: 0px 11px 37px rgba(35, 31, 32, 0.5);
      outline: none;

      &:hover {
        opacity: .8;
      }

      @media only screen and (max-width: 500px) {
        font-size: .8em;
      }

      @media only screen and (max-width: 350px) {
        font-size: .7em;
      }
    }
}
