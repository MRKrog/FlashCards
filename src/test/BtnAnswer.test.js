import React from "react";
import BtnAnswer from "../BtnAnswer";
import MockData from '../mockData';
import { shallow } from "enzyme";

const answer = ["Hyper Text Markup Language"]
const onChangeCard = jest.fn();


describe("Answer Button", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <BtnAnswer key={answer}
                 answer={answer}
                 onChangeCard={onChangeCard}
                 />
    );
  });

  it("should match snapshot when all data is passed correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should invoke the onChangeCard method when button is clicked", () => {
    wrapper.find("button").simulate("click", { preventDefault: () => {} });
    expect(onChangeCard).toBeCalled();
  });

});
