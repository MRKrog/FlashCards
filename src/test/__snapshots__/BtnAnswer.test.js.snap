// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answer Button should match snapshot when all data is passed correctly 1`] = `
<div
  className="BtnAnswer"
>
  <button
    onClick={[Function]}
  >
    Hyper Text Markup Language
  </button>
</div>
`;
